core/record/_record/recordUpdater.js

```diff
@@ -41,34 +41,32 @@ export const mergeNodes = nodes => record => {
   return recordUpdated
 }
 
-export const assocNodes = nodes => record => {
+export const assocNodes = (nodes) => (record) => {
   let recordUpdated = { ...record }
-  if (!(keys.nodes in recordUpdated)) {
-    recordUpdated[keys.nodes] = {}
-  }
+  recordUpdated[keys.nodes] = { ...RecordReader.getNodes(record) }
 
-  R.forEachObjIndexed((node, nodeUuid) => {
+  Object.entries(nodes).forEach(([nodeUuid, node]) => {
     if (Node.isDeleted(node)) {
       recordUpdated = deleteNode(node)(recordUpdated)
     } else {
       recordUpdated[keys.nodes][nodeUuid] = node
       recordUpdated = NodesIndex.addNode(node)(recordUpdated)
```

Comment:
what happens if the node previously exist?

Reply:
it's replaced by the passed one